pike/smb2.py

```diff
@@ -3843,3 +3843,48 @@ def _decode(self, cur):
         # TODO: handle the case where the buffer is too small
         self.snapshots = snapshot_buffer.strip("\0").split("\0")
 
+class SetZeroDataRequest(IoctlInput):
+    ioctl_ctl_code = IoctlCode.FSCTL_SET_ZERO_DATA
+
+    def __init__(self, parent):
+        IoctlInput.__init__(self, parent)
+        self.FileOffset = 0
+        self.BeyondFinalZero = 0
```

Comment:
pike style is to convert the names from the spec to all lowercase with capitals spaced with underscore:

`BeyondFinalZero` to `beyond_final_zero`